### DISCUSSION TITLE
User auth method phone number re-use

### DISCUSSION THREAD

**Discussion by merill** [maester365/maester#544] [Ideas]
Write a Maester check on phone numbers and find out user accounts where the same number is shared with more than three accounts. 

Would indicate users sharing MFA tokens.

**Comment by GuidoBaijense** [comment]
I've been working on this one. However, there isn't a good way to check this fast. We can leverage the authenticationmethods registration report to check for users with phoneMethods. But when we use batching to retrieve their registered PhoneNumbers, we don't get the associated userid back. I tried matching based on the index, but that isn't always correct.

If we want to show which user accounts have the same phoneNumber registered, we need good number/user association and we have to ditch the batch-endpoint. For small tenants, that's okay, but for large ones it will take hours to get all the numbers back.
